knox-homepage-ui/home/app/homepage.service.ts

```diff
@@ -27,8 +27,10 @@ import {SessionInformation} from './sessionInformation/session.information';
 
 @Injectable()
 export class HomepageService {
-    apiUrl = window.location.pathname.replace(new RegExp('home/.*'), 'api/v1/metadata/');
-    sessionUrl = window.location.pathname.replace(new RegExp('home/.*'), 'session/api/v1/sessioninfo');
+    pathParts = window.location.pathname.split('/');
+    topologyContext = '/' + this.pathParts[1] + '/' + this.pathParts[2] + '/';
+    apiUrl = this.topologyContext + 'api/v1/metadata/';
+    sessionUrl = this.topologyContext + 'session/api/v1/sessioninfo';
     generalProxyInformationUrl = this.apiUrl + 'info';
     publicCertUrl = this.apiUrl + 'publicCert?type=';
     topologiesUrl = this.apiUrl + 'topologies';
```

Comment:
Are these changes relevant to this patch, or is this just refactoring of the existing code?

Reply:
This is relevant. As you can see, other UIs were revised, and updated if necessary, to get their URLs relative to the topology context and not rely on hard-coded app names to be replaced.
Without this change the feature is not working.